src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -357,8 +357,13 @@ internal MethodSymbol Method
                         return analysis.Operator;
                     }
 
-                    if (uncommonData is DeconstructionUncommonData deconstruction
-                        && deconstruction.DeconstructMethodInfo.Invocation is BoundCall call)
+                    if (uncommonData is DeconstructionUncommonData {
+                        DeconstructMethodInfo:
+                        {
+                            Invocation: BoundCall { } call
```

Comment:
❔ Is there any need for the `{ }` here? @gafter What would you say is the canonical form for this pattern?

Reply:
In a declaration pattern (not using `var`), an empty set of curly braces is redundant as there is already a null check implicit in the type check.

---
In reply to: [344384091](https://github.com/dotnet/roslyn/pull/39738#discussion_r344384091) [](ancestors = 344384091)

Reply:
@alrz Action item here is to omit the empty pattern braces when the pattern also contains a type check.

Reply:
Verified fixed